plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/ApplicationSourceTreeItems.java

```diff
@@ -51,7 +52,15 @@
 	@SuppressWarnings({ "unchecked", "rawtypes" })
 	public <T> List<T> createChildren(Object parent) {
 		if (parent instanceof Connection) {
-			return (List<T>) ((Connection) parent).getResources(ResourceKind.PROJECT);
+		    List<T> activeProjects = new ArrayList<>();
+		    List<IResource> projectResources = ((Connection) parent).getResources(ResourceKind.PROJECT);
+		    for (IResource projectResource: projectResources) {
+		        IProject project = (IProject)projectResource;
+		        if (!"Terminating".equals(project.getStatus())) {
```

Comment:
I guess that the project state could be moved as constant to the client lib (we currently have lots of places where we use a "Running" to check for pod (and other resources) state)?

Reply:
@adietish https://github.com/openshift/openshift-restclient-java/pull/334

Reply:
merged and published the 6.1.0.Final client lib that contains your change: https://repository.jboss.org/nexus/service/local/repositories/releases/content/com/openshift/openshift-restclient-java/6.1.0.Final/openshift-restclient-java-6.1.0.Final.jar

Reply:
@bdshadow once you reference state checking from a string to the constant that's now available in the client lib (and of course start using the 6.1.0 client in manifest) I'll +1 this and have it merged.